### ISSUE TITLE
Installation of QuickSearch in Eclipse Oxygen fails due to missing dependency bundle com.google.guava

### ISSUE THREAD

**Issue by benjaminsailer** [spring-projects/spring-ide#318]
Hi there,

as the Eclipse Quicksearch plugin is a great feature but I am forced to use a plain eclipse oxygen release at our customer (without marketplace access, so installation via update site), I tried to get it using the eclipse-integration-commons-site.  Unfortunately this failed with the following error:

Cannot complete the install because one or more required items could not be found.
  Software being installed: Eclipse Quicksearch 3.9.5.201807030629-RELEASE (org.springsource.ide.eclipse.commons.quicksearch.feature.feature.group 3.9.5.201807030629-RELEASE)
  Missing requirement: Eclipse Integration Commands - Livexp 3.9.5.201807030629-RELEASE (org.springsource.ide.eclipse.commons.livexp 3.9.5.201807030629-RELEASE) requires 'bundle com.google.guava [15.0.0,16.0.0)' but it could not be found
  Cannot satisfy dependency:
    From: Quick Search 3.9.5.201807030629-RELEASE (org.springsource.ide.eclipse.commons.quicksearch 3.9.5.201807030629-RELEASE)
    To: bundle org.springsource.ide.eclipse.commons.livexp 0.0.0
  Cannot satisfy dependency:
    From: Eclipse Quicksearch 3.9.5.201807030629-RELEASE (org.springsource.ide.eclipse.commons.quicksearch.feature.feature.group 3.9.5.201807030629-RELEASE)
    To: org.springsource.ide.eclipse.commons.quicksearch [3.9.5.201807030629-RELEASE]

I'll attach the eclipse installation used.

[eclipse_installation.txt](https://github.com/spring-projects/spring-ide/files/2374404/eclipse_installation.txt)

Any help appreciated.  Thanks in advance,
Benjamin

**Comment by benjaminsailer** [comment]
Unfortunately I tried also this update site - same effect.  I'll attach three pictures regarding the installation workflow:
1. UpdateSite- and Bundle Selection
2. Progress (note that eclipse tries to use alternative update sites)
3. InstallationResult - Failure

![2_installationprogress](https://user-images.githubusercontent.com/41427262/45416371-1a69a380-b680-11e8-8c8b-666f47ccc3aa.PNG)
![3_installationresult](https://user-images.githubusercontent.com/41427262/45416372-1a69a380-b680-11e8-90f2-ba07c6e519dd.PNG)
![1_updatesiteinstallationselection](https://user-images.githubusercontent.com/41427262/45416373-1b023a00-b680-11e8-8fd8-6cd0cb73a7ed.PNG)

Is the guava bundle assumed to be part of the vanilla installation (and probably removed from my customers eclipse customization?)

**Comment by martinlippert** [comment]
You are right, the guava bundle is not part of our own update site, but instead usually fetched from the main Eclipse (Oxygen, in your case) update site. Do you have that update site in the preferences? I can see that you have the "contact all update sites" enabled, which should allow the install procedure to grab the guava piece from the main Oxygen site.

**Comment by benjaminsailer** [comment]
My bad I forgot to add the eclipse update site to my sites, sorry for that.

However, even adding http://download.eclipse.org/eclipse/updates/4.7 (and even 4.6) did not resolve the issue.  It seems that I already have installed guava but i a version 21.0.0.v20170206-1425, while the dependency of liveexp (and therefore quicksearch) seems to be fixed to [15,16).

Is there a reason for this narrow dependency?  Or should I try to get rid of guava in the new version and expect the old one to come along with the dependencies?

**Comment by martinlippert** [comment]
The release repository for Eclipse Oxygen is http://download.eclipse.org/releases/oxygen/, not the platform updates site that you mentioned. And I double checked, it contains the right version of Guava.

The reason for the narrow dependency definition are resolution conflict affects when STS components are installed together with the Buildship extension (and probably others, too). Therefore we had to constrain this dependency down to that version range.

This will change in future updates, since we already worked on refining that dependency, so I think future versions of STS 3.9.x will not require that guava version anymore.
